core/commonMain/src/ArgParser.kt

```diff
@@ -128,39 +144,59 @@ open class ArgParser(val programName: String, var useDefaultHelpShortName: Boole
     protected val fullCommandName = mutableListOf(programName)
 
     /**
-     * Origin of option/argument value.
-     *
-     * Possible values:
-     * SET_BY_USER - value of option was provided in command line string;
-     * SET_DEFAULT_VALUE - value of option wasn't provided in command line, but set using default value;
-     * UNSET - value of option is unset
-     * REDEFINED - value of option was redefined in source code after parsing.
+     * The way an option/argument has got its value.
      */
-    enum class ValueOrigin { SET_BY_USER, SET_DEFAULT_VALUE, UNSET, REDEFINED }
+    enum class ValueOrigin {
+        /* The value was parsed from command line arguments. */
+        SET_BY_USER,
+        /* The value was missing in command line, therefore the default value was used. */
+        SET_DEFAULT_VALUE,
+        /* The value is not yet initialized. */
+        UNSET,
+        /* The value was redefined manually using after parsing with the property setter. */
+        REDEFINED,
+    }
 
     /**
-     * Options prefix style.
-     *
-     * Possible values:
-     * LINUX - Linux style, for full forms of options "--", for short form - "-"
-     * JVM - JVM style, both for full and short forms of options "-"
+     * The style of option prefixing.
      */
-    enum class OPTION_PREFIX_STYLE { LINUX, JVM }
+    // TODO: make enum name pascal-cased
+    enum class OPTION_PREFIX_STYLE {
+        /* Linux style: the full name of an option is prefixed with two hyphens "--" and the short name — with one "-". */
+        LINUX,
+        /* JVM style: both full and short names are prefixed with one hyphen "-". */
+        JVM,
+    }
 
     /**
-     * Add option with single possible value and get delegator to its value.
+     * Declares a named option and returns an object which can be used to access the option value
+     * after all arguments are parsed or to delegate a property for accessing the option value to.
+     *
+     * By default, the option supports only a single value, is optional, and has no default value,
+     * therefore its value's type is `T?`.
      *
-     * @param type argument type, one of [ArgType].
-     * @param fullName argument full name.
-     * @param shortName option short name.
-     * @param description text description of Argument.
-     * @param deprecatedWarning text message with information in case if option is deprecated.
+     * You can alter the option properties by chaining extensions for the option type on the returned object:
+     *   - [AbstractSingleOption.default] to provide a default value that is used when the option is not specified;
+     *   - [SingleNullableOption.required] to make the option non-optional;
+     *   - [AbstractSingleOption.delimiter] to allow specifying multiple delimited values for the option;
```

Comment:
Multiple values here seems to be a bit confusing for me.
May be "to allow specifying several values joined by delimiter in one command line value "?

Reply:
Or, rephrased, but we need to come up with a more consistent terms for these things.